pylint/checkers/typecheck.py

```diff
@@ -1153,43 +1151,67 @@ def _check_assignment_from_function_call(self, node):
         if not isinstance(function_node, funcs):
             return
 
-        # Unwrap to get the actual function object
+        # Unwrap to get the actual function node object
         if isinstance(function_node, astroid.BoundMethod) and isinstance(
             function_node._proxied, astroid.UnboundMethod
         ):
             function_node = function_node._proxied._proxied
 
         # Make sure that it's a valid function that we can analyze.
         # Ordered from less expensive to more expensive checks.
-        # pylint: disable=too-many-boolean-expressions
         if (
             not function_node.is_function
-            or isinstance(function_node, nodes.AsyncFunctionDef)
-            or function_node.decorators
-            or function_node.is_generator()
-            or function_node.is_abstract(pass_is_abstract=False)
-            or utils.is_error(function_node)
-            or not function_node.root().fully_defined()
+            or function_node.decorators is not None
```

Comment:
In the original code we do `function_node.decorators`. Is there a difference? I believe decorators is `Union[None, nodes.Decorators]` so we could also just use `or function_node.decorators`